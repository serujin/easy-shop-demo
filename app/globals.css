@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
