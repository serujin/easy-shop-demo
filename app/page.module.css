.main {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.main * {
  display: flex;
}

.ratingsContainer {
  background-color: #f97805;
  width: 100%;
  padding: 2% 0;
  justify-content: center;
  align-items: center;
  color: white;
}

.productsContainer {
  margin-top: 2%;
  width: 52%;
}
