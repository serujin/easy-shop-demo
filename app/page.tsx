'use client'
import ItemWrapper from '@/components/basic/ItemWrapper'
import Ratings from '@/components/ratings/Ratings'
import { useEffect, useState } from 'react'
import { getAllProductsBasicInformation } from '@/lib/products'
import styles from './page.module.css'

const createProductHref = (id: string) => `/products/${id}`
const createProductImgSrc = (id: string) => `/products/${id}/icon.webp`

export default function Home() {
  const [products, setProducts] = useState([])

  useEffect(() => {
    getAllProductsBasicInformation().then((products) => setProducts(products))
  }, [])

  return (
    <main className={styles.main}>
      <div className={styles.headerContainer}>
        <div className={styles.ratingsContainer}>
          <Ratings
            title={'Custom stickers'}
            titleStyle="huge"
            rating={4.8}
            reviews={155761}
          />
        </div>
      </div>
      <div className={styles.productsContainer}>
        <div className={styles.productsWrapper}>
          {products.map(({ id, name }, index) => (
            <ItemWrapper
              key={index}
              href={createProductHref(id)}
              imgAlt={name}
              imgSrc={createProductImgSrc(id)}
              title={name}
            />
          ))}
        </div>
      </div>
    </main>
  )
}
