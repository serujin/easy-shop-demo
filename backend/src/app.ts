import express from 'express'
import jsonGraphqlExpress from 'json-graphql-server'
import cors from 'cors'
import data from '../database.json'

const app = express()

app.use(cors())
app.use('/', jsonGraphqlExpress(data))

export default app
