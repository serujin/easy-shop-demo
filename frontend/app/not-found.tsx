import NotFound from '@/components/basic/not-found/NotFound'

export default function Custom404() {
  return <NotFound is404 />
}
