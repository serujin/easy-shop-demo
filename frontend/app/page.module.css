.main {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.main * {
  display: flex;
}

.headerContainer {
  background-color: #f97805;
  width: 100%;
  align-items: center;
  justify-content: center;
}

.ratingsContainer {
  width: 52%;
  padding: 2% 0;
  justify-content: center;
  align-items: center;
  color: white;
}

.productsContainer {
  width: 100%;
  align-items: center;
  justify-content: center;
}

.productsWrapper {
  margin-top: 2%;
  width: 52%;
  flex-wrap: wrap;
}
