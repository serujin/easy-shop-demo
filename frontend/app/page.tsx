'use client'
import ItemWrapper from '@/components/basic/item-wrapper/ItemWrapper'
import Ratings from '@/components/ratings/Ratings'
import { useEffect, useState } from 'react'
import { getAllProductsBasicInformation } from '@/lib/products'
import styles from '@/styles/custom-stickers-page.module.css'
import NotFound from '@/components/basic/not-found/NotFound'

const createProductHref = (id: string) => `/products/${id}`
const createProductImgSrc = (id: string) => `/products/${id}/icon.webp`

export default function Home() {
  const [products, setProducts] = useState([])

  useEffect(() => {
    getAllProductsBasicInformation().then(products => setProducts(products))
  }, [])

  if (!products) {
    return <NotFound isIndex />
  }

  return (
    <main className={styles.main}>
      <div className={styles.headerContainer}>
        <div className={styles.ratingsContainer}>
          <Ratings
            title={'Custom stickers'}
            titleStyle="huge"
            rating={4.8}
            reviews={155761}
          />
        </div>
      </div>
      <div className={styles.productsContainer}>
        <div className={styles.productsWrapper}>
          {products &&
            products.map(({ id, name }, index) => (
              <ItemWrapper
                key={index}
                href={createProductHref(id)}
                imgAlt={name}
                imgSrc={createProductImgSrc(id)}
                title={name}
              />
            ))}
        </div>
      </div>
    </main>
  )
}
