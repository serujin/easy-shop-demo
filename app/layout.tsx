import NavBar from "@/components/nav-bar/NavBar";
import type { Metadata } from "next";
import "./globals.css";

export const metadata: Metadata = {
  title: "Easy shop demo",
  description: "Simple app to showcase and learn skills",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body>
        <header>
          <NavBar />
        </header>
        {children}
      </body>
    </html>
  );
}
