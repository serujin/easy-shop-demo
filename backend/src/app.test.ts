import { describe, it } from '@jest/globals'
import app from './app'
import request from 'supertest'
import database from '../database.json'

const correctData = {
  allProducts: database.products.map(product => ({
    id: product.id,
    name: product.name,
  })),
}

describe('Test the root path', () => {
  it('should response the POST method', () => {
    const data = {
      query: '{allProducts{id,name}}',
    }
    return request(app).post('/').send(data).expect(200)
  })
  it('should response the correct data', () => {
    const data = {
      query: '{allProducts{id,name}}',
    }
    return request(app)
      .post('/')
      .send(data)
      .then(response => {
        expect(response.body.data).toEqual(correctData)
      })
  })
  it('should response the correct data', () => {
    const data = {
      query: `{Product(id:"${correctData.allProducts.at(0)?.id}"){name,description,rating,reviews,sizeOptions,quantityOptions}}`,
    }
    return request(app)
      .post('/')
      .send(data)
      .then(response => {
        expect(response.body.data).toMatchSnapshot()
      })
  })
})
