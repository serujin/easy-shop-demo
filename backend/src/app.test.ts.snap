// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the root path should response the correct data 1`] = `
{
  "Product": {
    "description": "Custom die cut stickers are a fast and easy way to promote your business, brand or event. Thick, durable vinyl protects your stickers from scratches, water & sunlight.",
    "name": "Die cut stickers",
    "quantityOptions": [
      {
        "price": {
          "currency": "$",
          "quantity": 60,
        },
        "value": "50",
      },
      {
        "discount": 39,
        "price": {
          "currency": "$",
          "quantity": 73,
        },
        "value": "100",
      },
      {
        "discount": 60,
        "price": {
          "currency": "$",
          "quantity": 95,
        },
        "value": "200",
      },
      {
        "discount": 68,
        "price": {
          "currency": "$",
          "quantity": 115,
        },
        "value": "300",
      },
      {
        "discount": 75,
        "price": {
          "currency": "$",
          "quantity": 152,
        },
        "value": "500",
      },
      {
        "discount": 81,
        "price": {
          "currency": "$",
          "quantity": 232,
        },
        "value": "1000",
      },
      {
        "discount": 85,
        "price": {
          "currency": "$",
          "quantity": 371,
        },
        "value": "2000",
      },
      {
        "discount": 86,
        "price": {
          "currency": "$",
          "quantity": 496,
        },
        "value": "3000",
      },
      {
        "discount": 88,
        "price": {
          "currency": "$",
          "quantity": 723,
        },
        "value": "5000",
      },
      {
        "discount": 90,
        "price": {
          "currency": "$",
          "quantity": 1225,
        },
        "value": "10000",
      },
    ],
    "rating": 4.5,
    "reviews": 76746,
    "sizeOptions": [
      {
        "value": "2″ x 2″",
      },
      {
        "value": "3″ x 3″",
      },
      {
        "value": "4″ x 4″",
      },
      {
        "value": "5″ x 5″",
      },
    ],
  },
}
`;
